pkg/volume/util/nested_volumes.go

```diff
@@ -70,16 +71,18 @@ func getNestedMountpoints(name, baseDir string, pod v1.Pod) ([]string, error) {
 		}
 		return nil
 	}
-	for _, container := range pod.Spec.InitContainers {
-		if err := checkContainer(&container); err != nil {
-			return nil, err
-		}
-	}
-	for _, container := range pod.Spec.Containers {
-		if err := checkContainer(&container); err != nil {
-			return nil, err
+
+	var err error
+	podutil.VisitContainers(&pod.Spec, func(c *v1.Container) bool {
+		if err = checkContainer(c); err != nil {
```

Comment:
interestingly enough this might read easier as

```
err = checkContainer(c)
return err == nil
```

I won't necessarily force that, but if it reads easier and is less likely for someone to accidentally alias `if err := ...` which is a common bug